job_status.go

```diff
@@ -71,7 +71,7 @@ func IsTerminalState(state JobState) bool {
 }
 
 // GetJobStatus retrieves the current status of a Buildkite job with retry logic
-func GetJobStatus(client *buildkite.Client, org, pipeline, build, jobID string) (*JobStatus, error) {
+func GetJobStatus(client *buildkite.Client, ctx context.Context, org, pipeline, build, jobID string) (*JobStatus, error) {
```

Comment:
this is a nitpick but it is tweaking my golang OCD a bit that `ctx` isn't the first param here